src/swap/changehero.js

```diff
@@ -0,0 +1,412 @@
+
+  if (initOptions.apiKey == null) {
+    throw new Error('No ChangeHero apiKey or secret provided.')
+  }
+  const { apiKey } = initOptions
+
+  async function call(json: any) {
+    const body = JSON.stringify(json)
+
+    const headers = {
+      'Content-Type': 'application/json',
+      'api-key': apiKey
+    }
+    const response = await fetchCors(uri, { method: 'POST', body, headers })
+
+    if (!response.ok) {
+      throw new Error(`ChangeHero returned error code ${response.status}`)
+    }
+    return response.json()
+  }
+
+  const out: EdgeSwapPlugin = {
+    swapInfo,
+    async fetchSwapQuote(
+      request: EdgeSwapRequest,
+      userSettings: Object | void
+    ): Promise<EdgeSwapQuote> {
+      const fixedPromise = this.getFixedQuote(request, userSettings)
+      // Estimated swaps disabled
+      const fixedResult = await fixedPromise
+      return fixedResult
+    },
+    async getFixedQuote(
+      request: EdgeSwapRequest,
+      userSettings: Object | void
+    ): Promise<EdgeSwapQuote> {
+      const [fromAddress, toAddress] = await Promise.all([
+        getAddress(request.fromWallet, request.fromCurrencyCode),
+        getAddress(request.toWallet, request.toCurrencyCode)
+      ])
+      const quoteAmount =
+        request.quoteFor === 'from'
+          ? await request.fromWallet.nativeToDenomination(
+              request.nativeAmount,
+              request.fromCurrencyCode
+            )
+          : await request.toWallet.nativeToDenomination(
+              request.nativeAmount,
+              request.toCurrencyCode
+            )
+
+      const { safeFromCurrencyCode, safeToCurrencyCode } = safeCurrencyCodes(
+        CURRENCY_CODE_TRANSCRIPTION,
+        request
+      )
+
+      const fixedRateQuote = await call({
+        jsonrpc: '2.0',
+        id: 'one',
+        method: 'getFixRate',
+        params: {
+          from: safeFromCurrencyCode,
+          to: safeToCurrencyCode
+        }
+      })
+
+      const [{ id: responseId }] = fixedRateQuote.result
+      const params =
+        request.quoteFor === 'from'
+          ? {
+              amount: quoteAmount,
+              from: safeFromCurrencyCode,
+              to: safeToCurrencyCode,
+              address: toAddress,
+              extraId: null,
+              refundAddress: fromAddress,
+              refundExtraId: null,
+              rateId: responseId
+            }
+          : {
+              amountTo: quoteAmount,
+              from: safeFromCurrencyCode,
+              to: safeToCurrencyCode,
+              address: toAddress,
+              extraId: null,
+              refundAddress: fromAddress,
+              refundExtraId: null,
+              rateId: responseId
+            }
+
+      const sendReply = await call({
+        jsonrpc: '2.0',
+        id: 2,
+        method: 'createFixTransaction',
+        params
+      })
+      checkReply(sendReply, request)
+      const quoteInfo: FixedQuoteInfo = sendReply.result
```

Comment:
results from servers should use `cleaners` to check the payload at runtime. See

https://github.com/EdgeApp/edge-exchange-plugins/blob/master/src/swap/letsexchange.js#L38